src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -750,7 +750,15 @@ trait ContextErrors {
 
       // def stabilize
       def NotAValueError(tree: Tree, sym: Symbol) = {
-        issueNormalTypeError(tree, sym.kindString + " " + sym.fullName + " is not a value")
+        /* Give a better error message for `val thread = java.lang.Thread`.
+         * In general, saying "Java-defined {class,interface}" everywhere looks
+         * a bit ugly (usually we don't care), but here it avoids mentioning the
+         * compiler fiction that is Java companion modules. */
```

Comment:
Just a personal opinion: I wonder if this code comment works better as a commit comment?

It's kind of tabs-vs-spaces, where the question is should I see this explanation inline or when I consult the commit after git blame? There are precedents for explaining inline.

I'd also love it if tooling made it easy to discover the discussion on the PR. Everything is forever.

Reply:
I can follow your reasoning, but personally I like (reasonable) explanations in line. Commit messages cannot be updated over time if the code changes, so you often end up having to track down multiple commits. (In genearl - this is a small example of course, the code is not likely to change often over time).